### DISCUSSION TITLE
How to look up entity by other key

### DISCUSSION THREAD

**Discussion by leon** [salesforce/akita#545] [Q&A]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
[ ] Other... Please describe:
</code></pre>

## Question

I have an entity which has a database id which is the idKey i use for the store.
This makes it easy to get an entity by calling `getEntity(id)`

But I also have a orderId which I would like to be able to quickly look up entities by.

I can look up the entity by looping through all the whole store but that seems very bad performance wise.

I thought about adding a orderIds: Map<OrderId, EntityId> to the root of the store and use that to look up the id.
But I'm not sure how to best keep that map updated.

How would you solve this?

**Comment by NetanelBasal** [comment]
Can you show what you're trying to do, please?

**Comment by hakimio** [comment]
Store can only have one `idKey`, but you can use [store middleware functions](https://datorama.github.io/akita/docs/additional/middleware/) to keep your map updated.
Anyway, the slower option is to just use `getAll()` with `filterBy` option.
